src/main/java/com/hermesworld/ais/galapagos/topics/controller/TopicController.java

```diff
@@ -143,19 +143,22 @@ public void addProducerToTopic(@PathVariable String environmentId, @PathVariable
 
     }
 
-    @PostMapping(value = "/api/change-owner/{envId}/{topicName}/{producerApplicationId}")
+    @PostMapping(value = "/api/change-owner/{envId}/{topicName}")
     public void changeTopicOwner(@PathVariable String envId, @PathVariable String topicName,
-            @PathVariable String producerApplicationId) {
-        if (envId.isEmpty() || topicName.isEmpty() || producerApplicationId.isEmpty()) {
+            @RequestBody ChangeTopicOwnerDto request) {
+        if (envId.isEmpty() || topicName.isEmpty() || request.getProducerApplicationId().isEmpty()) {
```

Comment:
Note that `request` AND `request.getProducerApplicationId()` could both be `null`. Should be checked here.